### PR TITLE
tbi(odata-service-inquirer): Move `ErrorHandler` and `getCFAbapInstanceChoices` to allow re-use

### PR DESCRIPTION
**Pull request by IainSAP**
#2554 

- Moves `ErrorHandler` and i18n texts from `@sap-ux/odata-service-inquirer` -> `@sap-ux/inquirer-common`
- Move `getCFAbapInstanceChoices` and i18n texts rom `@sap-ux/odata-service-inquirer` -> `@sap-ux/inquirer-common`
- Adds tests to cover moved code that was previously covered with tested consuming code

- Put some structure on `fiori-generator-shared` to promote discovery and reuse

### CODE REVIEW

**Comment by IainSAP** [comment]
Note, since there are no functional or code changes to `@sap-ux/fiori-generator-common` it will not be version bumped.

**Review by cianmSAP — DISMISSED** [review]
Updates look good
Have not tested changes
Changeset is appropriate

File: packages/odata-service-inquirer/test/unit/prompts/service-url/validators.test.ts [hunk 1]
```diff
@@ -170,6 +170,8 @@ describe('Test service url validators', () => {
                 odataService,
                 'axiosConfig': {}
             })
-        ).toBe(t('errors.odataServiceUrlNotFound'));
+        ).toBe(
+            'The service URL you have provided is not a valid OData Service. Fiori applications require an OData service as the data source.'
```
Comment (longieirl) [inline packages/odata-service-inquirer/test/unit/prompts/service-url/validators.test.ts:174]: I think this should be `SAP Fiori applications`.

**Review by longieirl — APPROVED** [review]
Comments addressed.
Good test coverage.
No outstanding issues.

**Pull request merged**